### ISSUE TITLE
Set resource limits for containers

### ISSUE THREAD

**Issue by bigg01** [openshift/file-integrity-operator#68]
As a Platform Engineer I need to control usage of CPU Memory per Container. 
Please add ressource limits:

```
resources:
          requests:
            cpu: 100m
            memory: 100Mi
          limits:
            memory: 200Mi
            cpu: 400m
```

The Aide PODS were running for a day and used 1.6 Gb of memory for no reason.

cheers

**Comment by jhrozek** [comment]
Thank you for filing the issue. We'll look into it next sprint.
While the resource limits are something we wanted to set either way, we also want to see if we can find the root cause of the leak.

**Comment by felixkrohn** [comment]
Would it be possible to re-open this issue? After a week running the pods consume about 3GiB RAM each.
Current workaround could be to set namespaced defaults, but I find this less elegant.

**Comment by JAORMX** [comment]
@feli

**Comment by JAORMX** [comment]
@felixkrohn what versoin are you using?

**Comment by felixkrohn** [comment]
0.1.13 as distributed by RH on operatorhub (image: http://quay.io/file-integrity-operator/file-integrity-operator:0.1.13)

**Comment by felixkrohn** [comment]
Is there anything I can do to help you debug this? (we're not yet running it in production)
![image](https://user-images.githubusercontent.com/17177764/114354666-d4925600-9b6e-11eb-8985-f93d7accbaf2.png)

**Comment by JAORMX** [comment]
@felixkrohn we'll look into it.

**Comment by mrogers950** [comment]
@felixkrohn would you be able to run with the steps outlined in https://mrogers950.gitlab.io/openshift/2021/04/12/fio-profile/ ? 
It will enable pprof for the ds pods, but requires a container build from source. If you can capture the heap data at a few points (like once a few days in, then again the next week), that could be useful for us to take a look at. I've traced the same slow leak myself and it would be good to have a comparison.

**Comment by felixkrohn** [comment]
@mrogers950 Thanks to the great how-to 👍 I got it running, and will send you the .gz files next week (don't hesitate to remind me should I forget...)

**Comment by felixkrohn** [comment]
Did the traces help in any way?
Would it be OK to add memory limits (something between 500 and 1000M) to the f-i-o deployment, or do you expect this could cause unwanted side effects or even reduce reliability of the results?

**Comment by mrogers950** [comment]
@felixkrohn yes, thanks for your help! the pprof data shows what I expected, which is that the daemon's actual heap usage is only a small percentage of the total that is reported by the cluster:
Here only about 7mb total:
![image](https://user-images.githubusercontent.com/3697177/118705943-d216c200-b7e6-11eb-962b-eb653c407a9b.png)

This coincides with what I found about the reserved space used by the go runtime which, I tried to outline briefly here: https://mrogers950.gitlab.io/golang/2021/03/12/wild-crazy-golang-mem/
So I believe the high usage will be addressed by https://github.com/golang/go/issues/44167 , (referenced by https://github.com/golang/go/issues/43699). 

But I think that now we can support pod limits properly because the daemon pods are more robust and should be able to handle restart by OOM occasionally. I'll work on a PR for that.
